code/game/mecha/mecha_defense.dm

```diff
@@ -1,10 +1,10 @@
 /obj/mecha/proc/get_armour_facing(relative_dir)
 	switch(relative_dir)
-		if(0) // BACKSTAB!
+		if(180) // BACKSTAB!
 			return facing_modifiers[MECHA_BACK_ARMOUR]
-		if(45, 90, 270, 315)
```

Comment:
do these not matter for diagonal shots?

Reply:
its kind of hard to tell what part you're hitting when you include diagonals, would northeast be a side or front hit? and the same with southeast being a side or backstab? let me know how you want this to be handled, mostly side hits and only direct attacks being front or backstab?

Reply:
i hate how this works since we're passing a get_dir() to take_damage() instead of a get_angle() which would be a lot more precise, since get_dir() considers anything that didn't originate from directly on the same x / y as the mob is considered a diagonal attack even if they're only off by 1 square a screen away lol


think it would probably be best to maintain the current/old broken behavior of south being crit, n/ne/nw being front, and the rest being side armor.